src/lib/api.ts

```diff
@@ -37,10 +37,13 @@ export default function API( {
 } = {} ): ApolloClient< NormalizedCacheObject > {
 	const errorLink = onError( ( { networkError, graphQLErrors } ) => {
 		if ( networkError && 'statusCode' in networkError && networkError.statusCode === 401 ) {
-			console.error(
-				chalk.red( 'Unauthorized:' ),
-				'You are unauthorized to perform this request, please logout with `vip logout` then try again.'
-			);
+			let message =
+				'You are unauthorized to perform this request, please logout with `vip logout` then try again.';
+			if ( 'result' in networkError && networkError.result?.code === 'token-disabled-inactivity' ) {
```

Comment:
Is the `in` check actually necessary since we have `?`?

```suggestion
			if ( networkError.result?.code === 'token-disabled-inactivity' ) {
```

Reply:
@mjangda We need the `result in` to make TS aware we're using the class `ServerError` instead of `ServerParseError` from the Apollo Client.

```
import { ServerError } from '../utils';
import { ServerParseError } from '../http';
export interface ErrorResponse {
    graphQLErrors?: ReadonlyArray<GraphQLError>;
    networkError?: Error | ServerError | ServerParseError;
    [....]
}

If we use your code the type checker will error out with
```
 Property 'result' does not exist on type 'ServerParseError'.

42    if ( networkError.result?.code === 'token-disabled-inactivity' ) {
```

Reply:
Interesting, thank you for explaining that.

Maybe we should make the check more obvious to indicate that?

```suggestion
			if ( networkError instanceof ServerError && networkError.result?.code === 'token-disabled-inactivity' ) {
```

Reply:
This doesn't work either (I tested before landing with the `result in`) since we're checking against a type (ServerError) and not a class.

```
> tsc

src/lib/api.ts:43:33 - error TS2359: The right-hand side of an 'instanceof' expression must be either of type 'any', a class, function, or other type assignable to the 'Function' interface type, or an object type with a 'Symbol.hasInstance' method.

43    if ( networkError instanceof ServerError && networkError.result?.code === 'token-disabled-inactivity' ) {
                                   ~~~~~~~~~~~

src/lib/api.ts:43:61 - error TS2339: Property 'result' does not exist on type 'ServerError | ServerParseError'.
  Property 'result' does not exist on type 'ServerParseError'.

43    if ( networkError instanceof ServerError && networkError.result?.code === 'token-disabled-inactivity' ) {
                                                               ~~~~~~


Found 2 errors in the same file, starting at: src/lib/api.ts:43
```

What we can do, if we want to improve the readability, is to have a `isServerError` function with the same check @mjangda 

something like
```
function isServerError(networkError){
return 'result' in networkError;
}
```

Reply:
Even more interesting :)

I think the readability would be nice improvement but it's very minor.

Reply:
Added in 0d8eefed9cf6d6ce5729ca0d2b0eb88b2f1ed57e . For future reference here's also the docs around using `in` operator narrowing:  https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing